### ISSUE TITLE
Opening a new tab from a container window opens a normal tab and not a container tab

### ISSUE THREAD

**Issue by SoftVision-EmilPasca** [mozilla/testpilot-containers#245]
[Affected versions]:
- Firefox 51.0.1 and up
- Containers Experiment 0.9.3

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- Have Containers Experiment(v0.9.3) installed on a Firefox new profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisite.
2. Open any containers tab and navigate to any website.
3. Click on the "Containers" button in the toolbar.
4. Click the arrow next to the previously used container and choose to "Move tabs to a new window".
5. Press "CTRL+T" and observe the newly opened tab.

[Expected result]:
- A new tab is opened and is part of the same container.

[Actual result]:
- A normal new tab is opened.

[Notes]:
- After step 5 you no longer have the context-menu option to open tabs in that container.
![containers_window_issue](https://cloud.githubusercontent.com/assets/20083658/23220470/cd476f9c-f92a-11e6-954e-98d028dfec48.gif)

**Comment by johngruen** [comment]
This is as intended, but I have a feeling people will treat this as a bug, so let's keep it open.

**Comment by TanviHacks** [comment]
@johngruen What do you think about this?  This is like a Container Window feature.

**Comment by smichel17** [comment]
User here. I expected that when I opened a new tab, it would be the same container type as whatever my current tab was. Ie, follow @SoftVision-EmilPasca's steps but omit steps 3 and 4.

**Comment by nkestrel** [comment]
If you want to inherit the current tab's container you need to middle/ctrl + click the new tab button (Firefox 53+ [Bug 1325014](https://bugzilla.mozilla.org/show_bug.cgi?id=1325014)), this also creates the new tab next to the current tab. A normal click on the new tab button creates an unrelated tab at the end of the tabstrip and so should not inherit the current tab's container.

**Comment by smichel17** [comment]
I wrote a response to this, but then realized that we're trying to solve the wrong problem.

### The real problem here is that we force the user to pick a context *before* they enter the URL.

I seldom open a new tab thinking "I want to do something in [some container]!" I open a new tab thinking "I want to visit [some site]!" THEN, once I've opened the tab and typed the URL, I think about what container this belongs in.

Sometimes, I open a new tab, start to type something, then change my mind about what I want to open. Under the current workflow, I need to close this tab, open a new tab in the new context, and then type the new URL. That just doesn't work very well.

Here's a mockup I made in #311 that is also relevant here. I'd love to have this instead of the current awkward pop-up to choose a container for my new tab:

> ![firefox-containers-link](https://cloud.githubusercontent.com/assets/6563664/23574898/8a2b65ea-0052-11e7-9f10-d97ea3f023ad.png)

My original response:

> That's awesome functionality that I didn't know about. Is there a keyboard shortcut that does the same thing? I open most of my new tabs with \<Ctrl-t>.
> 
> I'd encourage the devs to add a notification of some sort the first time a user opens a new tab when they are currently in a container, letting them know how to open a new tab with the same context. Then track how often people choose to keep the same container vs start a new one.

**Comment by JohnLGalt** [comment]
I tested nkestrel's reply - a middle click on the **new tab button** does, indeed, open a new tab in the same container.  I'll have to double check that no TMP settings interfere with this, but so far, it looks like this is a good enough solution for me, as I make heavy use of left-, middle- and right-clicking in Firefox.

**Comment by jurf** [comment]
What about a keyboard shortcut to open a new tab in the current context?

Also middle clicking on the button does not open it in the same context for me.

**Comment by imyxh** [comment]
@jurf I think that would be ideal if the developers don't want a different shortcut for every container (alt+1, alt+2, etc.)
Maybe ctrl+shift+t to open a tab in the current container, and ctrl+t to open a tab in the regular container.

**Comment by smichel17** [comment]
> Maybe ctrl+shift+t to open a tab in the current container, and ctrl+t to open a tab in the regular container.

@imyxh <ctrl-shift-t> is already taken; it opens the most recently closed tab.

**Comment by imyxh** [comment]
@smichel17 Oh. Right. I use that all the time, come to think of it.
Though I'm sure there's another substitute available, like Alt + T or Ctrl + Alt + T (though the latter opens a terminal in most distributions of Linux.)

**Comment by grahamperrin** [comment]
**Alt-T** = **T**ools menu (Firefox on KDE Plasma 4 on FreeBSD)

Points of reference that may be useful include: 

- [Mac keyboard shortcuts - Apple Support](https://support.apple.com/HT201236)
- https://docs.kde.org/trunk4/en/kde-runtime/fundamentals/kbd.html
- https://docs.kde.org/stable5/en/kde-workspace/plasma-desktop/shortcuts.html
- in Ask Ubuntu, the [accepted answer](http://askubuntu.com/a/122776/25036) to _Enabling native shortcuts on Gnome - 3_ 

– and so on; generally, I should refrain from predefined shortcuts. Allow the user to set a preference, if he or she finds for a shortcut.

**Comment by lmorchard** [comment]
Not sure if this issue is progressing at all, but:

> The real problem here is that we force the user to pick a context before they enter the URL.

For my use, I *do* want this. I want every URL in a window to be in the same container. If I hit Ctrl-T from an existing container tab, I want that new tab to be in the same container. Middle-click on the new tab button works, but I *never* use the mouse in opening new tabs - my flow is Ctrl-T, start typing URL or partial search terms

**Comment by smichel17** [comment]
> For my use, I do want this. I want every URL in a window to be in the same container. If I hit Ctrl-T from an existing container tab, I want that new tab to be in the same container. Not exactly the same things, but I think the latter makes the former easier to maintain.

@Imorchard I don't think our views are in conflict here. You just want the default selection to be the same as your current container. (For the record, I would prefer that, too, or in private tabs like #429.)

**Comment by lelandbatey** [comment]
Could a setting be added that makes it so new tabs inherit the container they where "opened from"?

By default it would be disabled, so typing `ctrl+t` in a container tab `Personal` would open a tab in the default container, which is no container. Once the setting where enabled, if `ctrl+t` where entered in a `Personal` container tab, the newly created tab would also be a `Personal` container tab.

I feel like making this a setting is a reasonable way to provide users the power to choose their interaction style without interrupting the new-tab process. Thoughts?

**Comment by Symbai** [comment]
I would support the idea of it being optional instead of a new hotkey or its forced to anything.

**Comment by groovecoder** [comment]
Duplicate of https://github.com/mozilla/testpilot-containers/issues/462 which has more up-votes. Please transfer any up-votes there (if you haven't already).

**Comment by Ithiloneth** [comment]
**Tree Style Tab** by [Piro](https://addons.mozilla.org/en-US/firefox/user/9480/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_content=addons-manager-user-profile-link) has a function that lets you automate this for opening a new tab.

It works me with:

- Ctrl+t
- Scroll-wheel (middle mouse) click on link
-  "+" button  in Tree Style Tab
- And more I'm sure; only _just now_ set this up as a solution to this issue

![image](https://user-images.githubusercontent.com/20944949/110211660-ead63380-7e97-11eb-9888-d73c818a2e68.png)

_My version of TST is 3.5.34 - I run a legacy version due to not liking a an update to how groups for tabs are created with a recent update._
